pkg/controller/appdefinition/generation.go

```diff
@@ -16,7 +17,11 @@ func UpdateObservedFields(req router.Request, resp router.Response) error {
 
 func impliedAutoUpgrade(appSpec v1.AppInstanceSpec) bool {
```

Comment:
This is most likely no longer needed. Holding off on deleting for the moment until I know for sure that we don't want something keeping track of this at a Status level.

Reply:
Cancel that, I think it is actually still necessary since the situation where a AutoUpgrade pattern is set still counts for AutoUpgrade being enabled even if `app.Spec.AutoUpgrade` is not. 

Additionally, we use this status field in the table printer for Apps to show if AutoUpgrade is enabled.